src/main/webapp/feed.html

```diff
@@ -21,11 +21,48 @@ <h1 class="title">🌏project title</h1>
         <div id="events-feed"></div>
         <div class="feed-right-side">
             <div class="feed-box create-event-box"><div class="feed-box-content create-event-box-content"><p class="icon">📅</p><p class="create-event-text">create an event</p></div></div>
-            <div class="feed-box challenges-box">
+            <div class="feed-box challenges-box" onclick="openChallengesModal()">
                 <div class="feed-box-content challenges-box-content">
                     <p class="icon">🌏</p><p class="challenges-text">challenges</p>
                 </div>
-                <div class="challenges-badge" id="challenges-badge"><p class="icon badge-icon" id="challenges-badge-icon"></p><p class="challenge-steps" id="challenge-steps"></p></div>
+                <div class="challenges-badge" id="challenges-badge">
+                    <p class="icon badge-icon" id="challenges-badge-icon"></p>
+                    <p class="challenge-steps" id="challenge-steps"></p>
+                    <div class="badge-filling" id="feed-badge-filling"></div>
+                </div>
+            </div>
+        </div>
+    </div>
+    <div id="challenges-modal">
+        <div id="challenges-nav-bar" class="challenges-modal-content">
+            <p id="challenges-nav-bar-header">Challenges</p>
+        </div>
+        <div id="challenges-main-panel" class="challenges-modal-content">
+            <p id="close-modal-button" onclick="closeChallengesModal()">x</p>
+            <div id="challenges-main-panel-upper-content">
+                <div id="challenges-main-panel-text" class="challenges-main-panel-text">
+                    <p id="challenges-main-panel-header" class="challenges-main-panel-content"></p>
+                    <p id="challenges-main-panel-step" class="challenges-main-panel-content"></p>
+                    <p id="challenges-main-panel-description" class="challenges-main-panel-content"></p>
```

Comment:
Also, from line 43 - 48 ish, I suggest adding a "challenges-main-panel-content" wrapper div. That way, we eliminate the repeated lines.
This will also allow use to traverse the elements inside . 
css:
**_challenges-main-panel-content > p {
...
}_**
Now you can affect all the elements inside your wrapper(as long as they are at the same "depth" i believe)

Also I would suggest using div elements and have the paragraph element inside. 
ex: 
  **_div id= "challenges-main-panel-header"
     p information /p 
  /div_**

This would change your css to be 
**_challenges-main-panel-content > div {
...
}_**